util/db/dbutil.go

```diff
@@ -386,6 +389,9 @@ func (db *Accessor) GetPageSize(ctx context.Context) (pageSize uint64, err error
 
 // dbretry returns true if the error might be temporary
 func dbretry(obj error) bool {
+	if err := errors.Unwrap(obj); err != nil { // unwrap error if wrapped
```

Comment:
I think you could use `errors.As(err, sqlite3.Error)` to avoid the warning about wrapping the inner error.